internal/persistence/postgres/pool_test.go

```diff
@@ -0,0 +1,120 @@
+package postgres
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/kong/koko/internal/log"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+func TestPoolOpts(t *testing.T) {
+	t.Run("pool options are applied to the pgxpool.Config", func(t *testing.T) {
+		d := 42 * time.Second
+		opt := Opts{
+			DBName:   "koko",
+			Hostname: "localhost",
+			Port:     DefaultPort,
+			User:     "koko",
+			Password: "koko",
+			Pool: PoolOpts{
+				MaxConns:          42,
+				MinConns:          42,
+				MaxConnLifetime:   d,
+				MaxConnIdleTime:   d,
+				HealthCheckPeriod: d,
+			},
+		}
+		logger := log.Logger
+		pgxPoolConfig, err := NewPgxPoolConfig(opt, logger)
+		require.NoError(t, err)
+		require.Equal(t, int32(42), pgxPoolConfig.MaxConns)
+		require.Equal(t, int32(42), pgxPoolConfig.MinConns)
+		require.Equal(t, d, pgxPoolConfig.MaxConnLifetime)
+		require.Equal(t, d, pgxPoolConfig.MaxConnIdleTime)
+		require.Equal(t, d, pgxPoolConfig.HealthCheckPeriod)
+	})
+	t.Run("pool options are validated", func(t *testing.T) {
+		d := 10 * time.Second
+		poolOpts := []PoolOpts{
+			{
+				MaxConns:          0,
+				HealthCheckPeriod: d,
+			},
+			{
+				MaxConns:          -1,
+				HealthCheckPeriod: d,
+			},
+			{
+				MaxConns:          10,
+				HealthCheckPeriod: 0,
+			},
+			{
+				MaxConns:          10,
+				HealthCheckPeriod: -1,
+			},
+		}
+		for _, poolOpt := range poolOpts {
+			opt := Opts{
+				DBName:   "koko",
+				Hostname: "localhost",
+				Port:     DefaultPort,
+				User:     "koko",
+				Password: "koko",
+				Pool:     poolOpt,
+			}
+
+			err := opt.Validate()
+			require.Error(t, err)
+		}
+	})
+}
+
+func TestPoolRegistration(t *testing.T) {
```

Comment:
Maybe add a test case for an already registered pool?

Reply:
+1, will do

Reply:
done